site/src/pages/DeploySettingsPage/OAuth2AppsSettingsPage/EditOAuth2AppPageView.tsx

```diff
@@ -0,0 +1,308 @@
+
+        {isLoadingApp && <Loader />}
+
+        {!isLoadingApp && app && (
+          <>
+            <DeleteDialog
+              isOpen={showDelete}
+              confirmLoading={isUpdating === "delete-app"}
+              name={app.name}
+              entity="OAuth2 application"
+              onConfirm={() => deleteApp()}
+              onCancel={() => setShowDelete(false)}
+            />
+
+            <Stack direction="row">
+              <div>
+                <strong>Client ID:</strong>
+              </div>
+              <CopyableValue value={app.id}>{app.id}</CopyableValue>
+            </Stack>
+
+            <Divider css={{ borderColor: theme.palette.divider }} />
+
+            <OAuth2AppForm
+              app={app}
+              onSubmit={updateApp}
+              isUpdating={isUpdating === "update-app"}
+              error={error}
+              actions={
+                <Button
+                  variant="outlined"
+                  type="button"
+                  color="error"
+                  onClick={() => setShowDelete(true)}
+                >
+                  Delete&hellip;
+                </Button>
+              }
+            />
+
+            <Divider css={{ borderColor: theme.palette.divider }} />
+
+            <OAuth2AppSecretsTable
+              secrets={secrets}
+              generateAppSecret={generateAppSecret}
+              deleteAppSecret={deleteAppSecret}
+              isLoadingSecrets={isLoadingSecrets}
+              isUpdating={isUpdating}
+            />
+          </>
+        )}
+      </Stack>
+    </>
+  );
+};
+
+type OAuth2AppSecretsTableProps = {
+  secrets?: TypesGen.OAuth2AppSecret[];
+  generateAppSecret: () => void;
+  isLoadingSecrets: boolean;
+  isUpdating:
+    | "update-app"
+    | "create-secret"
+    | "delete-app"
+    | "delete-secret"
+    | false;
+  deleteAppSecret: (id: string) => void;
+};
+
+const OAuth2AppSecretsTable: FC<OAuth2AppSecretsTableProps> = ({
+  secrets,
+  generateAppSecret,
+  isLoadingSecrets,
+  isUpdating,
+  deleteAppSecret,
+}) => {
+  return (
+    <>
+      <Stack
+        alignItems="baseline"
+        direction="row"
+        justifyContent="space-between"
+      >
+        <h2>Client secrets</h2>
+        <LoadingButton
+          loading={isUpdating === "create-secret"}
+          type="submit"
+          variant="contained"
+          onClick={() => generateAppSecret()}
+        >
+          Generate secret
+        </LoadingButton>
+      </Stack>
+
+      <TableContainer>
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableCell width="80%">Secret</TableCell>
+              <TableCell width="20%">Last Used</TableCell>
+              <TableCell width="1%" />
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {isLoadingSecrets && <TableLoader />}
+            {!isLoadingSecrets && (!secrets || secrets?.length === 0) && (
```

Comment:
Could the second case be simplified down to `!secrets?.length`? I think that would catch both the undefined and 0 cases

Reply:
It definitely could, for some reason I like to always compare `length` to  a number rather than make use of falsey checks.

Reply:
Honestly, that's generally what I default to, too – I guess my main concern here was making sure the condition isn't too long, since there's not an easy way to name it without moving the value all the way above the return statement